api/token.go

```diff
@@ -44,9 +50,45 @@ type RefreshTokenGrantParams struct {
 	RefreshToken string `json:"refresh_token"`
 }
 
+// IdTokenGrantParams are the parameters the IdTokenGrant method accepts
+type IdTokenGrantParams struct {
+	IdToken  string `json:"id_token"`
+	Nonce    string `json:"nonce"`
+	Provider string `json:"provider"`
+}
+
 const useCookieHeader = "x-use-cookie"
 const useSessionCookie = "session"
 
+func (p *IdTokenGrantParams) getVerifier(ctx context.Context) (*oidc.IDTokenVerifier, error) {
+	config := getConfig(ctx)
+	var provider *oidc.Provider
+	var err error
+	var clientId string
+	switch p.Provider {
+	case "apple":
+		clientId = config.External.Apple.ClientID
```

Comment:
should we check if each provider is explicitly currently enabled in the config?

Reply:
just in case someone wants to toggle it off

Reply:
good idea! will add this